cli/cmd/install_helm_test.go

```diff
@@ -0,0 +1,207 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"testing"
+
+	"github.com/linkerd/linkerd2/pkg/charts"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"k8s.io/helm/pkg/chartutil"
+	pb "k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/renderutil"
+	"sigs.k8s.io/yaml"
+)
+
+func TestRenderHelm(t *testing.T) {
+	var (
+		chartName      = "linkerd2"
+		goldenFileName = "install_helm_output.golden"
+		ha             = false
+		namespace      = "linkerd-dev"
+	)
+
+	// read the control plane chart and its defaults from the local folder.
+	// override certain defaults with pinned values.
+	// use the Helm lib to render the templates.
+	// the golden file is generated using the following `helm template` command:
+	// helm template --set Identity.TrustAnchorsPEM="test-crt-pem" --set Identity.Issuer.TLS.CrtPEM="test-crt-pem" --set Identity.Issuer.TLS.KeyPEM="test-key-pem" charts/linkerd2  --set Identity.Issuer.CrtExpiry="Jul 30 17:21:14 2020" --set ProxyInjector.KeyPEM="test-proxy-injector-key-pem" --set ProxyInjector.CrtPEM="test-proxy-injector-crt-pem" --set ProfileValidator.KeyPEM="test-profile-validator-key-pem" --set ProfileValidator.CrtPEM="test-profile-validator-crt-pem" --set Tap.KeyPEM="test-tap-key-pem" --set Tap.CrtPEM="test-tap-crt-pem" --set LinkerdVersion="linkerd-version"  > cli/cmd/testdata/install_helm_output.golden
```

Comment:
Yeah, I have to manually reorder the `helm template` YAML output. I *think* it's because `renderutil.Render()` returns a `map`. So whenever there is a `for` loop either in our code or the Helm lib to iterate that map, the order of the YAML changes. Hence, I'm not able to use something like the [FakeClient.InstallReleaseFromChart()](https://godoc.org/k8s.io/helm/pkg/helm#FakeClient.InstallReleaseFromChart) to generate the YAML in the test (i.e. less code for us), because the ordering will always be [different](https://github.com/helm/helm/blob/df5ca23a418199c1b2af3abfc420f8339d0b9b8e/pkg/helm/fake.go#L384-L390).

I added this comment to keep a record of all the `--set` options used to generate the golden file.